pinot-core/src/main/java/org/apache/pinot/core/query/pruner/ColumnValueSegmentPruner.java

```diff
@@ -94,60 +103,159 @@ public List<IndexSegment> prune(List<IndexSegment> segments, QueryContext query)
     }
     FilterContext filter = Objects.requireNonNull(query.getFilter());
     ValueCache cachedValues = new ValueCache();
+    Map<String, DataSource> dataSourceCache = new HashMap<>();
     int numSegments = segments.size();
     List<IndexSegment> selectedSegments = new ArrayList<>(numSegments);
-    if (query.isEnablePrefetch()) {
-      FetchContext[] fetchContexts = new FetchContext[numSegments];
-      try {
-        // Prefetch bloom filter for columns within the EQ/IN predicate if exists
-        for (int i = 0; i < numSegments; i++) {
-          IndexSegment segment = segments.get(i);
-          FetchContext fetchContext = _fetchPlanner.planFetchForPruning(segment, query);
-          if (!fetchContext.isEmpty()) {
-            segment.prefetch(fetchContext);
-            fetchContexts[i] = fetchContext;
-          }
+    if (!query.isEnablePrefetch()) {
```

Comment:
Suggest making a separate pruner for bloom filter. Bloom filter can be applied only if there is EQ or IN with less than `_inPredicateThreshold` values. We should probably always use parallel pruning for bloom filter since it involves loading index.

Reply:
make sense!